### PR TITLE
Yhteydenottotiedot ohjaamoon

### LINKED ISSUE #6035: Yhteydenottotiedot ohjaamoon
Yhteydenottotiedot ohjaamoon. Opiskelijan ohjaajalle täydet tiedot, muille obfuskoidut

### PR DESCRIPTION
**Pull request by saaralepisto**
Closes #6035

### CODE REVIEW

**Comment by saaralepisto** [comment]
https://github.com/otavanopisto/pyramus/pull/1352

File: muikku-elastic-search/src/main/java/fi/otavanopisto/muikku/plugins/search/ElasticSearchProvider.java [hunk 1]
```diff
@@ -843,6 +843,41 @@ private String prepareQueryString(String queryString) {
     prepared = prepared.replace(" ", "* ");
     return prepared + "*";
   }
+  
+  @Override
+  public SearchResult findUserGroup(SchoolDataIdentifier identifier) {
+
+    // Query that checks activity based on user having a study end date set
```
Comment (pkukkon) [inline muikku-elastic-search/src/main/java/fi/otavanopisto/muikku/plugins/search/ElasticSearchProvider.java:850]: This comment doesn't relate to this method at all?

**Review by pkukkon — CHANGES_REQUESTED** [review]

File: muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUserSchoolDataBridge.java [hunk 2]
```diff
@@ -1200,7 +1202,94 @@ public BridgeResponse<List<OrganizationContactPerson>> listOrganizationContactPe
...
+  
+  @Override 
+  public BridgeResponse<StudentContactLogEntryRestModel> createStudentContactLogEntry(SchoolDataIdentifier studentIdentifier, StudentContactLogEntryRestModel payload){
+    Long studentId = identifierMapper.getPyramusStudentId(studentIdentifier.getIdentifier());
+    
+    if (studentId == null) {
+      logger.severe(String.format("Student for identifier %s not found", studentIdentifier));
+      return null;
+    }
+    return pyramusClient.responsePost(String.format("/students/students/%d/contactLogEntries", studentId), Entity.entity(payload, MediaType.APPLICATION_JSON), StudentContactLogEntryRestModel.class);
```
Comment (pkukkon) [inline muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUserSchoolDataBridge.java:1235]: The creatorId in all response objects (contact log entries and their comments) are ids of the users in Pyramus. Before handing the response over to Muikku front-end, they should be converted to ids of the users in Muikku (i.e. userEntityId). Something like...

```
if (response.getEntity() != null && response.getEntity().getCreatorId() != null) {
  SchoolDataIdentifier identifier = identifierMapper.getStaffIdentifier(response.getEntity().getCreatorId());
  if (identifier != null) {
    UserSchoolDataIdentifier usdi = userSchoolDataIdentifierController.findUserSchoolDataIdentifierBySchoolDataIdentifier(identifier);
    if (usdi != null) {
      response.getEntity.setCreatorId(usdi.getUserEntity.getId());
    }
    else {
      // TODO log warning
      response.getEntity.setCreatorId(null);
    }
  }
  else {
    // TODO log warning
    response.getEntity.setCreatorId(null);
  }
}
// TODO Do the same with comments
```

Naturally this applies to all methods returning contact log entries and/or their comments.

**Review by pkukkon — CHANGES_REQUESTED** [review]

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java [hunk 3]
```diff
@@ -688,6 +697,216 @@ private GuiderStudentWorkspaceActivityRestModel toRestModel(GuiderStudentWorkspa
...
+   * text: "something something",
+   * creatorId: 23,
+   * creatorName: "Etunimi Sukunimi";
+   * entryDate: 2021-02-15;
+   * type: "PHONE";
+   * }
+   */
+  @POST
+  @Path("/student/{ID}/contactLog/")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java:726]: No permission handling within the method at all. Perhaps...

`@RESTPermit (GuiderPermissions.ACCESS_CONTACT_LOG)`

...given to all staff roles. Same with all the other endpoints dealing with contact log entries.

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java [hunk 4]
```diff
@@ -688,6 +697,216 @@ private GuiderStudentWorkspaceActivityRestModel toRestModel(GuiderStudentWorkspa
...
+   */
+  
+  @PUT
+  @Path("/student/{ID}/contactLog/{CONTACTLOGENTRYID}")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response updateStudentContactLog(@PathParam("ID") Long userEntityId, @PathParam("CONTACTLOGENTRYID") Long contactLogEntryId, StudentContactLogEntryRestModel payload) {
+    String dataSource = sessionController.getLoggedUserSchoolDataSource();
+    UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+    
+    BridgeResponse<StudentContactLogEntryRestModel> response = userSchoolDataController.updateStudentContactLogEntry(dataSource, userEntity.defaultSchoolDataIdentifier(), contactLogEntryId, payload);
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java:763]: Extra safety by checking that the creatorId in the payload object is that of the logged in user?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java [hunk 5]
```diff
@@ -688,6 +697,216 @@ private GuiderStudentWorkspaceActivityRestModel toRestModel(GuiderStudentWorkspa
...
+   * Output: Updated StudentContactLogEntryCommentRestModel
+   */
+  @PUT
+  @Path("/student/{ID}/contactLog/{ENTRYID}/comments/{COMMENTID}")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response updateStudentContactLogComment(@PathParam("ID") Long userEntityId, @PathParam("ENTRYID") Long entryId, @PathParam("COMMENTID") Long commentId, StudentContactLogEntryCommentRestModel payload) {
+    String dataSource = sessionController.getLoggedUserSchoolDataSource();
+    UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+    
+    BridgeResponse<StudentContactLogEntryCommentRestModel> response = userSchoolDataController.updateStudentContactLogEntryComment(dataSource, userEntity.defaultSchoolDataIdentifier(), entryId, commentId, payload);
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java:880]: Same as when updating a contact log entry; check that the creatorId in payload is the same as the id of the currently logged in user.

**Review by pkukkon — CHANGES_REQUESTED** [review]

**Review by pkukkon — APPROVED** [review]

**Review by pkukkon — APPROVED** [review]

**Review by aviljakaine — APPROVED** [review]

**Pull request merged**